typescript/cli/src/tests/core-apply.e2e-test.ts

```diff
@@ -0,0 +1,127 @@
+import { expect } from 'chai';
+import { Signer, Wallet, ethers } from 'ethers';
+
+import { ProxyAdmin__factory } from '@hyperlane-xyz/core';
+import {
+  CoreConfig,
+  ProtocolFeeHookConfig,
+  randomAddress,
+} from '@hyperlane-xyz/sdk';
+import { Address } from '@hyperlane-xyz/utils';
+
+import { readYamlOrJson, writeYamlOrJson } from '../utils/files.js';
+
+import {
+  hyperlaneCoreApply,
+  hyperlaneCoreDeploy,
+  readCoreConfig,
+} from './commands/core.js';
+import {
+  ANVIL_KEY,
+  CHAIN_NAME_2,
+  CORE_CONFIG_PATH,
+  CORE_READ_CONFIG_PATH_2,
+  DEFAULT_E2E_TEST_TIMEOUT,
+  REGISTRY_PATH,
+} from './commands/helpers.js';
+
+describe('hyperlane core apply e2e tests', async function () {
+  this.timeout(2 * DEFAULT_E2E_TEST_TIMEOUT);
+
+  let signer: Signer;
+  let initialOwnerAddress: Address;
+
+  before(async () => {
+    const chainMetadata: any = readYamlOrJson(
```

Comment:
why any?

Reply:
Just copy pasted from the old test, fixed it to use the `ChainMetadata` type!